File: RelationExample/RelationExample/Building.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelationExample
{
    class Building
    {
        private Room[] room;

        public Building(Room[] room)
        {
            this.room = room;
        }

        public float CalculateVolume()
        {
            float volume = 0;
            for (int i = 0; i < room.Length; i++)
            {
                volume += room[i].Area * room[i].Height;
            }
            return volume;
        }

        public void PrintRoomSize(float size)
        {
            Console.WriteLine("The volume of the building is: " + size);
        }
    }
}

File: RelationExampleC++/RelationExampleC++/Building.cpp
#include "Building.h"

Building::Building(std::vector<Room> rooms)
{
	m_rooms = rooms;
}

float Building::CalculateVolume()
{
	float volume = 0;
	for (int i = 0; i < m_rooms.size(); i++)
	{
		volume += m_rooms[i].GetArea() * m_rooms[i].GetHeight();
	}
	return volume;
}

void Building::PrintRoomSize(float size)
{
	std::cout << "The volume of the room of the building is : " << size;
}